wordpress-mobile/WordPress-Android
# Adding reblog dedicated tracking events.

This PR adds specific tracking events for reblog.

The dedicated events added are:

**reader_article_reblogged** -> triggered when reblogging from a list of posts in the reader (main unfiltered reader or a filtered list obtained by using the subfilter bar or tapping on a post site title)
**reader_article_detail_reblogged** -> triggered when reblogging from a reader opened post details

We were already traking in `editor_post_created` event `"created_post_source_detail":"post-from-reader-reblog"` property and we added a dedicated `"created_post_source_detail":"post-from-reader-detail-reblog"` to discriminate between the actions described above.

NOTE: I've set the `SiteModel site` parameter in the `openEditorForReblog` to @Nullable since it can be null for example in `ReaderPostDetailFragment > onActivityResult` when we do `siteStore.getSiteByLocalId(siteLocalId)`; managed it the same as a nullable post in the function.

## To test
- Activate tracking event in app settings
- Trigger a reblog from a list of post (unfiltered or filtered)
- Check in logcat you get something like

```
Tracked: reader_article_reblogged, Properties: {"target_blog_id":157782556,"source_post_id":14871,"source_blog_id":23519412}
Tracked: editor_post_created, Properties: {"blog_id":157782556,"created_post_source_detail":"post-from-reader-reblog","post_type":"post","created_post_source":"post-list","is_jetpack":false}
```

- Trigger a reblog from a post detail
- Check in logcat you get something like

```
Tracked: reader_article_detail_reblogged, Properties: {"target_blog_id":157782556,"source_post_id":2682331778,"source_blog_id":50586720}
Tracked: editor_post_created, Properties: {"blog_id":157782556,"created_post_source_detail":"post-from-reader-detail-reblog","post_type":"post","created_post_source":"post-list","is_jetpack":false}
```

PR submission checklist:
